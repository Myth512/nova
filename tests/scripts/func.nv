def printPlusOne(n) {
	print(n + 1)
}

print(printPlusOne(1))

def double(n) {
	return n * 2
}

print(double(5))

def pow(n, e) {
	return n ^ e
}

print(pow(2, 3))

def factorial(n) {
	if n == 0 {
		return 1
	}
	return n * factorial(n - 1)
}

print(factorial(5))

def fib(n) {
	if n < 2 {
		return n
	}
	return fib(n - 2) + fib(n - 1)
}

print(fib(20))

var := "im global"

def func() {
	print(var)
	var := "im local"
	print(var)
}

func()

def func_with_default_args(x=5, y=10, z=15) {
	print(x, y, z)
}

func_with_default_args()
func_with_default_args(1)
func_with_default_args(1, 2)
func_with_default_args(1, 2, 3)
