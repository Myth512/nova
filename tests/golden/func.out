2
nil
10
8
120
6765
im global
im local
5 10 15
1 10 15
1 2 15
1 2 3
